File: Seminar2/python/Classes/tax_inspector.py
from Classes.ordinary_client import OrdinaryClient
from Interfaces.i_actor_behaviour import iActorBehaviour


class TaxInspector(iActorBehaviour):
    def __init__(self):
        self.name = "Tax audit"
        self.isTakeOrder = False
        self.isMakeOrder = False


    def getName(self):
        return self.name

    def isTakeOrder(self):
        return self.isTakeOrder

    def isMakeOrder(self):
        return self.isMakeOrder

    def setTakeOrder(self, take):
        self.isTakeOrder = take

    def setMakeOrder(self, make):
        self.isMakeOrder = make

    def getActor(self):
        return OrdinaryClient(self.name).getName()

File: Seminar2/java/SuperMarketApp/src/Classes/TaxInspector.java
package Classes;

import Interfaces.iActorBehaviour;

public class TaxInspector implements iActorBehaviour {
    private String name;
    private boolean isTakeOrder;
    private boolean isMakeOrder;
    

    public TaxInspector() {
        this.name = "Tax Audit";
    }

    public String getName()
    {
        return name;
    }

  
    public boolean isTakeOrder() {
        return isTakeOrder;
    }

    public boolean isMakeOrder() {
        return isMakeOrder; 
    }

    public void setTakeOrder(boolean val) {
        isTakeOrder = val;
    }
    public void setMakeOrder(boolean val) {
        isMakeOrder = val;        
    }

    @Override
    public Actor geActor() {
        return new OrdinaryClient(name);
    }
    
}
